### PR TITLE
submit channel

### PR DESCRIPTION
**Pull request by samparsky**
fixes issues #44 #30

### CODE REVIEW

File: routes/channel.js [hunk 1]
```diff
@@ -108,6 +114,32 @@ function getValidatorMessages(req, res, next){
 	.catch(next)
 }
 
+function createCampaign(req, res, next) {
+	const { id, depositAmount, depositAsset, validators, spec, watcher } = req.body
+	const channelCol = db.getMongo().collection('channel')
+	const channel = {
+		'_id': id,
```
Comment (Ivshti) [inline routes/channel.js]: use no quotation marks

File: routes/schema.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            "validators" : _validators(cfg)
+        }),
+        "watcher": Joi.object({
+            "ethereum": Joi.object({
+                "contract": Joi.string().required()
+            })
+        })
+    }),
+    validateCampaign: (cfg) => ({
+        "id" : Joi.string().required(),
```
Comment (Ivshti) [inline routes/schema.js]: no quottion marks again; maybe we can configure jslint to enforce this

File: routes/channel.js (hunk 1 above)
Comment (samparsky) [reply routes/channel.js]: done

File: routes/schema.js (hunk 2 above)
Comment (samparsky) [reply routes/schema.js]: done

File: routes/schema.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+				})
+				.required(),
+			fee: Joi.number().required()
+		})
+	)
+
+	if (CREATORS_WHITELIST && CREATORS_WHITELIST.length > 0) {
+		schema = schema.has(
+			Joi.object({
+				id: Joi.any().valid(CREATORS_WHITELIST),
```
Comment (Ivshti) [inline routes/schema.js]: that should be VALIDATORS_WHITELIST

File: routes/channel.js [hunk 4]
```diff
@@ -115,6 +125,27 @@ function getValidatorMessages(req, res, next) {
 		.catch(next)
 }
 
+function createCampaign(req, res, next) {
```
Comment (Ivshti) [inline routes/channel.js]: rename this, as well as the routes, to `createChannel` and POST `/channel`

File: test/fixtures.js [hunk 5]
```diff
@@ -0,0 +1,129 @@
+const cfg = {
+	CREATORS_WHITELIST: ['0xa95743F561db3618D204C9a7c3ca55cDf0625107'],
+	MINIMAL_DEPOSIT: 1000,
+	TOKEN_ADDRESS_WHITELIST: ['0x0e6BFF21862858a289AB214009d572b4079C8515']
+}
+
+module.exports = {
+	createCampaign: [
```
Comment (Ivshti) [inline test/fixtures.js]: rename to createChannel

File: routes/schema.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+		.items(Joi.string())
+		.required()
+		.length(2)
+	if (CREATORS_WHITELIST && CREATORS_WHITELIST.length > 0) {
+		schema = schema.has(CREATORS_WHITELIST)
+	}
+	return schema
+}
+
+function validators({ CREATORS_WHITELIST }) {
```
Comment (Ivshti) [inline routes/schema.js]: also, other than `VALIDATORS_WHITELIST` (separate thing than CREATORS_WHITELIST), this should check if the validator count is always 2; not larger, not smaller, always `=== 2`

and when checking the whitelist, we should check VALIDATORS_WHITELIST OR `=== adapter.whoami()`

File: test/fixtures.js (hunk 5 above)
Comment (samparsky) [reply test/fixtures.js]: done

File: routes/schema.js (hunk 3 above)
Comment (samparsky) [reply routes/schema.js]: done

File: routes/schema.js (hunk 6 above)
Comment (samparsky) [reply routes/schema.js]: yeah it already checks the validator count is always 2.

**Pull request merged**